### DISCUSSION TITLE
pyOptSparse Error: There was an error importing the compiled ffsqp module

### DISCUSSION THREAD

**Discussion by DanSJohnson** [mdolab/pyoptsparse#304] [Installation] [answered]
Hi everyone, I'm having some trouble with the FSQP optimizer. My code is just:
```
import pyoptsparse
opt=pyoptsparse.FSQP()
```

this results in the error:
```
+------------------------------------------------------------------------------+
| pyOptSparse Error: There was an error importing the compiled ffsqp module    |
+------------------------------------------------------------------------------+
```
Would anyone be able to advise on what i could do to fix this problem?

My operating system is Ubuntu 18.04.6 LTS and I am using pyoptsparse V2.1.0 with scipy version 1.2.3 and numpy version  1.19.5. I am not using conda, and these are all installed in a python virtual environment.

I installed pyoptsparse by activating my virtual env and executing the following commands:
```
git clone https://github.com/mdolab/pyoptsparse
cd pyoptsparse/
git checkout -b desired_version v2.1.0
pip install -e . --user
```

This appears to install pyoptsparse correctly, as I can import it into my code without issue, but still results in the above error when i try to use FSQP. This error is not present for most of the solvers in pyoptsparse, I was only able to reproduce it with IPOPT, NLQLP, ParOpt, SNOPT and FSQP. I think this is happening for the others because they require external code to operate which cannot be distributed with pyoptsparse for licensing reasons, is this also the case with FSQP? I had assumed FSQP would work "out of the box" because when i go to `/pyoptsparse/pyoptsparse/pyFSQP` there appear to be source files of some kind, though I may be misunderstanding what I'm looking at.

Any advice would be greatly appreciated!

**Comment by sseraj** [comment, marked as answer]
FSQP is proprietary like NLPQLP and SNOPT. The files we had in `pyFSQP/source` were not the actual source files. We removed FSQP in v2.1.2 because we could not test the wrapper without access to the source.

**Reply by DanSJohnson** [reply]
I see, thanks so much for your help!

Do you know of any other way to use FSQP from python? I'm trying to reproduce an academic paper and in their methodology they describe using FSQP. I could use an alternative, but it would be damaging to the integrity of the science as I wouldn't be representing the original work accurately.

**Reply by ewu63** [reply]
FSQP was always a proprietary software so you will have to contact the authors of the optimizer for licensing information. We do not have access to this software anymore which is why we removed the wrapper some time ago.
